test/benchmark/reasoner/iam/basic/BasicTest.java

```diff
@@ -0,0 +1,224 @@
+        benchmark.assertRunningTime(1500);
+        benchmark.assertCounters(10, NOBJECTS * NOBJECTS, 2, 2);
+    }
+
+    @Test
+    public void testRelationWithoutRolePlayer() {
+        String query = "match\n" +
+                "$r isa object-pair;\n";
+        Benchmark benchmark = new Benchmark("relation-without-rp", query, NOBJECTS * NOBJECTS);
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(1500);
+        benchmark.assertCounters(10, NOBJECTS * NOBJECTS, 2, 2);
+    }
+
+    @Test
+    public void testBoundRelation() {
+        String query = String.format(
+                "match\n" +
+                        "$o isa object, has id \"%s\";\n" +
+                        "(start: $o, end: $x) isa object-pair;\n",
+                queryParams.basicTestObject);
+        Benchmark benchmark = new Benchmark("relation-bound", query, NOBJECTS);
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(200);
+        benchmark.assertCounters(10, NOBJECTS, 2, 2);
+    }
+
+    @Test
+    public void testRelationUnpacking() {
+        String query = String.format(
+                "match\n" +
+                        "$o1 isa object, has id \"%s\";\n" +
+                        "$r1 (start: $o1, end: $x1) isa object-pair;\n" +
+                        "$r1 is $r2;\n" +
+                        "$r2 (start: $o2, end: $x2);\n",
+                queryParams.basicTestObject);
+        Benchmark benchmark = new Benchmark("relation-unpacking", query, NOBJECTS);
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(200);
+        benchmark.assertCounters(10, NOBJECTS, 2, NOBJECTS + 3);
+    }
+
+    @Test
+    public void testInferredRelationAndOwnership() {
+        String query = "match\n" +
+                "$r isa object-pair, has id $id;\n";
+        Benchmark benchmark = new Benchmark("inferred-relation-inferred-ownership", query, NOBJECTS * NOBJECTS * 2 - NOBJECTS);
+        benchmarker.runBenchmark(benchmark);
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(NOBJECTS * NOBJECTS + (NOBJECTS * NOBJECTS * 2 - NOBJECTS));
+        benchmark.assertCounters(25, 8060, 3, 3);
+    }
+
+    // More complicated
+    @Test
+    public void testDoubleJoin() {
+        String query = String.format(
+                "match\n" +
+                        "$a isa object, has id \"%s\";\n" +
+                        "(start: $a, end: $b) isa object-pair;\n" +
+                        "(start: $b, end: $c) isa object-pair;\n",
+                queryParams.basicTestObject);
+        Benchmark benchmark = new Benchmark("double-join", query, NOBJECTS * NOBJECTS);
+        benchmarker.runBenchmark(benchmark);
+
+        benchmark.assertAnswerCountCorrect();
+        benchmark.assertRunningTime(5000);
```

Comment:
looks like we're at 2 milliseconds per answer here again huh

Reply:
Yes, but also no.
There's still too much variance from the warmup for this to be indicative of the actual performance.
I'm just setting the bounds loosely till I get warmup to work better